torchgeo/trainers/segmentation.py

```diff
@@ -148,43 +150,38 @@ def configure_models(self) -> None:
                 param.requires_grad = False
 
     def configure_losses(self) -> None:
-        """Initialize the loss criterion.
-
-        Raises:
-            ValueError: If *loss* is invalid.
-        """
-        loss: str = self.hparams['loss']
-        ignore_index = self.hparams['ignore_index']
-        if loss == 'ce':
-            ignore_value = -1000 if ignore_index is None else ignore_index
-            self.criterion = nn.CrossEntropyLoss(
-                ignore_index=ignore_value, weight=self.hparams['class_weights']
-            )
-        elif loss == 'jaccard':
-            # JaccardLoss requires a list of classes to use instead of a class
-            # index to ignore.
-            classes = [
-                i for i in range(self.hparams['num_classes']) if i != ignore_index
-            ]
-
-            self.criterion = smp.losses.JaccardLoss(mode='multiclass', classes=classes)
-        elif loss == 'focal':
-            self.criterion = smp.losses.FocalLoss(
-                'multiclass', ignore_index=ignore_index, normalized=True
-            )
-        else:
-            raise ValueError(
-                f"Loss type '{loss}' is not valid. "
-                "Currently, supports 'ce', 'jaccard' or 'focal' loss."
-            )
+        """Initialize the loss criterion."""
+        ignore_index: int | None = self.hparams['ignore_index']
+        match self.hparams['loss']:
+            case 'ce':
+                ignore_value = -1000 if ignore_index is None else ignore_index
+                self.criterion: nn.Module = nn.CrossEntropyLoss(
+                    ignore_index=ignore_value, weight=self.hparams['class_weights']
+                )
+            case 'bce':
+                self.criterion = nn.BCEWithLogitsLoss()
```

Comment:
Is there any way we can modify this to support `ignore_index`? Maybe clever use of `pos_weight`?

Reply:
Probably would need to write a simple wrapper to do the masking of the loss. Something like this should work

```python
class BCE(nn.BCEWithLogitsLoss):
   def __init__(self, ignore_index, *args, **kwargs):
      super().__init__(*args, **kwargs)
      self.ignore_index = ignore_index
   
   def forward(self, input, target):
      ignore_mask = target == ignore_index
      loss = F.binary_cross_entropy_with_logits(
         input,
         target,
         self.weight,
         pos_weight=self.pos_weight,
         reduction='none'
      )
      loss = loss[~ignore_mask]

      if self.reduction = 'mean':
         loss = loss.mean()
      elif self.reduction = 'sum':
         loss = loss.sum()
      else:
         pass

      return loss
```